routes/tasks.js

```diff
@@ -15,6 +15,77 @@ router.get('/', function(req, res, next) {
 });
 
 
+/* GET tasks listing for feed, aggregated with user info. */
+router.get('/feed', function(req, res, next) {
+ req.app.locals.tasks.aggregate([
+   { $lookup:
+       {
+         from: 'users',
+         localField: 'user_id',
+         foreignField: '_id',
+         as: 'userDetails'
+       }
+   }
+ ]).toArray()
+   .then(result => {
+     res.setHeader('Content-Type', 'application/json');
+     res.status(200).send(JSON.stringify(result));
+   })
+   .catch(error => {
+     console.error(error);
+   })
+});
+
+/* PUT task: add clap to task */
+router.put('/feed/claps/:task_id', (req, res, next) => {
+  const task_id = ObjectID(req.params.task_id);
+  const { value, donor } = req.body;
+  const donor_id = ObjectID(donor);
+  req.app.locals.tasks.updateOne(
+    { _id: task_id },
+    {
+      $inc: { clapsReceived: value }
+    },
+    {
+      $push : { givenClaps: donor_id }
+    }
+  ).then((result) => {
+    res.status(200).end();
+  }).catch(err => {
+    console.error(err);
+    res.status(503).end();
+  });
+
+});
+
+/* PUT task: add mangos to task */
+router.put('/feed/mangos/:task_id', (req, res, next) => {
+  const task_id = ObjectID(req.params.task_id);
+  const { numMango, donor } = req.body;
+  const donor_id = ObjectID(donor);
+  const newMangoTransaction = {
+    user_id: donor_id,
+    mangoCount: numMango,
+    timestamp: new Date()
```

Comment:
should be Date.now()

Reply:
Unless new Date() is now by default - im not sure how it constructs

Reply:
https://github.com/drudge/mongoose-timestamp/issues/23

Should use Date.now() to be accurate